core/src/main/java/io/parsingdata/metal/data/ByteStreamSource.java

```diff
@@ -28,18 +29,22 @@
 
     public final ByteStream input;
 
-    public ByteStreamSource(final ByteStream input) {
+    ByteStreamSource(final ByteStream input) {
         this.input = checkNotNull(input, "input");
     }
 
     @Override
-    protected byte[] getData(final long offset, final BigInteger length) throws IOException {
-        if (!isAvailable(offset, length)) { throw new IOException("Data to read is not available ([offset=" + offset + ";length=" + length + ";source=" + this + ")."); }
-        return input.read(offset, length.intValue());
+    protected byte[] getData(final long offset, final BigInteger length) {
+        if (!isAvailable(offset, length)) { throw new RuntimeException("Data to read is not available ([offset=" + offset + ";length=" + length + ";source=" + this + ")."); }
```

Comment:
I'd rather not use plain `RuntimeException`s. In this case `IllegalStateException` is appropriate, I think?